tests/app/v2/notifications/test_post_notifications.py

```diff
@@ -82,34 +82,42 @@ def mock_deliver_sms(mocker):
 
 
 @pytest.mark.parametrize("reference", [None, "reference_from_client"])
-def test_post_sms_notification_returns_201(client, sample_template_with_placeholders, mock_deliver_sms, reference):
-    data = {
-        'phone_number': '+16502532222',
+@pytest.mark.parametrize("data", [
+    {"phone_number": "+16502532222"},
+    # TODO - Testing recipient_identifier requires an active feature flag that is not
+    # active in the testing environment.
+    # {"recipient_identifier": {"id_type": IdentifierType.VA_PROFILE_ID.value, "id_value": "bar"}},
+])
+def test_post_sms_notification_returns_201(client, sample_template_with_placeholders,
+                                           mock_deliver_sms, reference, data):
+    data.update({
         'template_id': str(sample_template_with_placeholders.id),
         'personalisation': {' Name': 'Jo'}
-    }
-    if reference:
-        data.update({"reference": reference})
+    })
+    if reference is not None:
+        data["reference"] = reference
 
     response = post_send_notification(client, sample_template_with_placeholders.service, 'sms', data)
+
     assert response.status_code == 201
-    resp_json = json.loads(response.get_data(as_text=True))
+    resp_json = response.get_json()
```

Comment:
Is this supposed to be `get_json()` or just `json()`?

Reply:
it's using `flask.response.get_json()`, which is proper according to the docs. It looks like you could also just use `.json()`, but that will just call the `.get_json()` method.
https://flask.palletsprojects.com/en/1.1.x/api/?highlight=response%20get_json#flask.Response.get_json